### ISSUE TITLE
Weak презентер сохраняется, если быстро вернуться на фрагмент

### ISSUE THREAD

**Issue by nikialeksey** [Arello-Mobile/Moxy#185]
[Вот этот gist](https://gist.github.com/nikialeksey/169c433bba22f4e213f5810db4263eee) содержит код и гифку происходящего. Если сделать презентер локальным - то все работает как часики. Может быть, я не понимаю, что значит `Weak` презентер? Однако данное поведение мне показалось подозрительным. Помогите разобраться, пожалуйста!

**Comment by xanderblinov** [comment]
Weak презентеры хранятся по weak ссылкам и удаляются при соответствующем проходе GC. Собственно, поэтому поведение такое странное

Просто не используйте Weak презентеры. В следующих релизах этот deprecate функционал будет удален

**Comment by nikialeksey** [comment]
Спасибо!

**Comment by senneco** [comment]
@xanderblinov нет, weak презентеры не хранятся по weak ссылкам. Не вводи людей в заблуждение ;)

**Comment by nikialeksey** [comment]
Тааак))) из за чего все-таки такое поведение? и правильно, что не нужно больше использовать Weak презенторы? Вот написано в документации к `PresenterType.WEAK`:

> Weak presenters are available everywhere. Weak presenter will be destroyed when finished all views. Inject will create new presenter instance.

тут сказано, что если вьюху убить, то презентер тоже умрет. Однако, гист, указанный в описании, показывает, что это не так.

**Comment by senneco** [comment]
Да, надо посмотреть, что не так. Но в чём Саша был прав -- weak (как и global) не ситоит использовать. Нужно идти по пути clean arch. Должна быть одна общая модель, а не один переиспользуемый презентер.

**Comment by nikialeksey** [comment]
Ок, спасибо! Может, тогда открыть эту `issue`?

**Comment by xanderblinov** [comment]
@senneco делай в следующий раз все через пул реквесты, плиз, а то ты хэшмапу выпилил в [коммите](https://github.com/Arello-Mobile/Moxy/commit/2b4a8c67225940a400022da79d1fcf4109842358#diff-9126fa8b205f3e8f93e4d32ef754df58) и теперь все сломалось. А я об этом и не подозревал :'(

**Comment by senneco** [comment]
@nikialeksey да, можно переоткрыть.
